samltest.py

```diff
@@ -284,7 +290,8 @@ def __init__(self, sp_url, logout_service, return_to):
                                                   return_to)
 
     def check_from_reply(self, reply):
-        pass
+        if reply.status_code == 400:
+            raise LogoutReplyError('Did not expect reply of Bad Request')
```

Comment:
I'd probably remove this function, the original intent (as you showed me in the mail you sent me) was that you should end up at the logout URL eventually. But Here you're testing a negative where you want 400 to be returned from mellon.
(And as discussed, for the malformed codes, you only expect 400).
If you want the code to be generic, I think testing for code > 300 could be OK too

Reply:
I just included the 400 check with the intention of adding more if it was useful to raise different exceptions for different replies.  So, in the interest in making sure it's at least raising an error for the logout I'll switch it to > 300 for a generic check.